diff --git a/data/calamares_polkit.orig b/data/calamares_polkit
index a76a1b9..08bfb09 100644
--- a/data/calamares_polkit.orig
+++ b/data/calamares_polkit
@@ -1,4 +1,6 @@
 #!/bin/bash
+NO_OOM_KILLER="`pidof -xw Xwayland` $$"
+echo $NO_OOM_KILLER | xargs -n1 sudo choom -n -1000 -p 
 if [ $(which pkexec) ]; then
 	pkexec --disable-internal-agent env DBUS_SESSION_BUS_ADDRESS="$DBUS_SESSION_BUS_ADDRESS" DISPLAY="$DISPLAY" GNOME_DESKTOP_SESSION_ID="$GNOME_DESKTOP_SESSION_ID" HOME="$HOME" KDE_FULL_SESSION="$KDE_FULL_SESSION" KDE_SESSION_VERSION="$KDE_SESSION_VERSION" QT_STYLE_OVERRIDE="$QT_STYLE_OVERRIDE" QT_QPA_PLATFORMTHEME="$QT_QPA_PLATFORMTHEME" QT_XCB_FORCE_SOFTWARE_OPENGL="$QT_XCB_FORCE_SOFTWARE_OPENGL" XAUTHORITY="$XAUTHORITY" XDG_CURRENT_DESKTOP="$XDG_CURRENT_DESKTOP" "/usr/bin/calamares" "-D6" "$@"
 else
