From 98d534d5dd814cd4940d05369038716467a4aa45 Mon Sep 17 00:00:00 2001
From: Frede Hundewadt <noreply@wonky.codeberg.org>
Date: Sun, 26 Nov 2023 14:57:19 +0100
Subject: [PATCH] [initcpiocfg] use f-string - use new style bash array (issue
 #2243)

---
 src/modules/initcpiocfg/main.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/modules/initcpiocfg/main.py b/src/modules/initcpiocfg/main.py
index 1d61369b95..0939054fb8 100644
--- a/src/modules/initcpiocfg/main.py
+++ b/src/modules/initcpiocfg/main.py
@@ -125,11 +125,11 @@ def write_mkinitcpio_lines(hooks, module
             # Replace HOOKS, MODULES and FILES lines with what we
             # have found via find_initcpio_features()
             if line.startswith("HOOKS"):
-                line = "HOOKS=\"{!s}\"".format(' '.join(hooks))
+                line = f"HOOKS=({str.join(' ', hooks)})"
             elif line.startswith("MODULES"):
-                line = "MODULES=\"{!s}\"".format(' '.join(modules))
+                line = f"MODULES=({str.join(' ', modules)})"
             elif line.startswith("FILES"):
-                line = "FILES=\"{!s}\"".format(' '.join(files))
+                line = f"FILES=({str.join(' ', files)})"
             mkinitcpio_file.write(line + "\n")
 
 
