From 2368c3dba226afce53cc301adaec4a8c5360ec98 Mon Sep 17 00:00:00 2001
From: Philip Mueller <philm@manjaro.org>
Date: Mon, 4 Dec 2023 06:37:57 +0700
Subject: [PATCH] [data] add env to polkit launcher

---
 data/calamares_polkit | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/data/calamares_polkit b/data/calamares_polkit
index 9bce0f82f4..b25df93af0 100644
--- a/data/calamares_polkit
+++ b/data/calamares_polkit
@@ -1,6 +1,11 @@
 #!/bin/bash
 if [ $(which pkexec) ]; then
-	pkexec --disable-internal-agent "/usr/bin/calamares" "$@"
+	pkexec --disable-internal-agent \
+	env DISPLAY="$DISPLAY" \
+	    XAUTHORITY="$XAUTHORITY" \
+	    DBUS_SESSION_BUS_ADDRESS="$DBUS_SESSION_BUS_ADDRESS" \
+	    HOME="$HOME" \
+	    "/usr/bin/calamares" "-D6" "$@"
 else
-	/usr/bin/calamares "$@"
+	/usr/bin/calamares "-D6" "$@"
 fi
-- 
GitLab

