From 4eac6fa8dcfab565399a3ddf97bb43121cbbf164 Mon Sep 17 00:00:00 2001
From: Philip Mueller <philm@manjaro.org>
Date: Mon, 4 Dec 2023 08:25:48 +0700
Subject: [PATCH] [data] polkit has to be one line

---
 data/calamares_polkit | 14 +-------------
 1 file changed, 1 insertion(+), 13 deletions(-)

diff --git a/data/calamares_polkit b/data/calamares_polkit
index db9e92de6..a32f1c451 100644
--- a/data/calamares_polkit
+++ b/data/calamares_polkit
@@ -1,18 +1,6 @@
 #!/bin/bash
 if [ $(which pkexec) ]; then
-	pkexec --disable-internal-agent \
-	env DBUS_SESSION_BUS_ADDRESS="$DBUS_SESSION_BUS_ADDRESS" \
-	    DISPLAY="$DISPLAY" \
-	    GNOME_DESKTOP_SESSION_ID="GNOME_DESKTOP_SESSION_ID" \
-	    HOME="$HOME" \
-	    KDE_FULL_SESSION="$KDE_FULL_SESSION" \
-	    KDE_SESSION_VERSION="$KDE_SESSION_VERSION" \
-            QT_STYLE_OVERRIDE="$QT_STYLE_OVERRIDE" \
-            QT_QPA_PLATFORMTHEME="$QT_QPA_PLATFORMTHEME" \
-            QT_XCB_FORCE_SOFTWARE_OPENGL="$QT_XCB_FORCE_SOFTWARE_OPENGL" \
-	    XAUTHORITY="$XAUTHORITY" \
-	    XDG_CURRENT_DESKTOP="XDG_CURRENT_DESKTOP" \ 
-	"/usr/bin/calamares" "-D6" "$@"
+	pkexec --disable-internal-agent env DBUS_SESSION_BUS_ADDRESS="$DBUS_SESSION_BUS_ADDRESS" DISPLAY="$DISPLAY" GNOME_DESKTOP_SESSION_ID="GNOME_DESKTOP_SESSION_ID" HOME="$HOME" KDE_FULL_SESSION="$KDE_FULL_SESSION" KDE_SESSION_VERSION="$KDE_SESSION_VERSION" QT_STYLE_OVERRIDE="$QT_STYLE_OVERRIDE" QT_QPA_PLATFORMTHEME="$QT_QPA_PLATFORMTHEME" QT_XCB_FORCE_SOFTWARE_OPENGL="$QT_XCB_FORCE_SOFTWARE_OPENGL" XAUTHORITY="$XAUTHORITY" XDG_CURRENT_DESKTOP="XDG_CURRENT_DESKTOP" "/usr/bin/calamares" "-D6" "$@"
 else
 	/usr/bin/calamares "-D6" "$@"
 fi
-- 
GitLab

