From 75ff3ee69c177e040d0e632907d3c27ea0923926 Mon Sep 17 00:00:00 2001
From: Philip Mueller <philm@manjaro.org>
Date: Thu, 30 Nov 2023 17:46:31 +0700
Subject: [PATCH] [modules] add kms to default hooks

---
 src/modules/initcpiocfg/main.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/modules/initcpiocfg/main.py b/src/modules/initcpiocfg/main.py
index 1fd4af714..eaff7e8e5 100644
--- a/src/modules/initcpiocfg/main.py
+++ b/src/modules/initcpiocfg/main.py
@@ -144,7 +144,7 @@ def find_initcpio_features(partitions, root_mount_point):
 
     :return 3-tuple of lists
     """
-    hooks = ["base", "udev", "autodetect", "modconf", "block", "keyboard", "keymap", "consolefont"]
+    hooks = ["base", "udev", "autodetect", "kms", "modconf", "block", "keyboard", "keymap", "consolefont"]
     modules = []
     files = []
 
-- 
GitLab

