From 8c873e0f49cef09a83a26c4ffc073925e1a91d4d Mon Sep 17 00:00:00 2001
From: Vitor Lopes <codesardine@users.noreply.github.com>
Date: Fri, 29 Jul 2022 18:03:44 +0100
Subject: [PATCH] add HDPI support

---
 src/calamares/main.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/calamares/main.cpp b/src/calamares/main.cpp
index 495127881..80a2b3a62 100644
--- a/src/calamares/main.cpp
+++ b/src/calamares/main.cpp
@@ -107,6 +107,7 @@ handle_args( CalamaresApplication& a )
 int
 main( int argc, char* argv[] )
 {
+    QApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
     CalamaresApplication a( argc, argv );
 
     KAboutData aboutData( "calamares",
-- 
GitLab

