diff --git a/src/calamares/CalamaresApplication.cpp b/src/calamares/CalamaresApplication.cpp
index 88e8e2919..a72e0863d 100644
--- a/src/calamares/CalamaresApplication.cpp
+++ b/src/calamares/CalamaresApplication.cpp
@@ -269,10 +269,11 @@ CalamaresApplication::initViewSteps()
 {
     cDebug() << "STARTUP: loadModules for all modules done";
     m_moduleManager->checkRequirements();
-    if ( Calamares::Branding::instance()->windowMaximize() )
+    if ( Calamares::Branding::instance()->windowFullscreen() )
     {
         m_mainwindow->setWindowFlag( Qt::FramelessWindowHint );
-        m_mainwindow->showMaximized();
+        m_mainwindow->setWindowFlag( Qt::WindowStaysOnTopHint );
+        m_mainwindow->showFullscreen();
     }
     else
     {
diff --git a/src/libcalamaresui/Branding.h b/src/libcalamaresui/Branding.h
index a5a2e535d..908693e81 100644
--- a/src/libcalamaresui/Branding.h
+++ b/src/libcalamaresui/Branding.h
@@ -219,7 +219,7 @@ public:
 
     bool welcomeStyleCalamares() const { return m_welcomeStyleCalamares; }
     bool welcomeExpandingLogo() const { return m_welcomeExpandingLogo; }
-    bool windowMaximize() const { return m_windowExpansion == WindowExpansion::Fullscreen; }
+    bool windowFullscreen() const { return m_windowExpansion == WindowExpansion::Fullscreen; }
     bool windowExpands() const { return m_windowExpansion != WindowExpansion::Fixed; }
     QPair< WindowDimension, WindowDimension > windowSize() const
     {
